### ISSUE TITLE
Support locking a file to prevent from deleting/moving

### ISSUE THREAD

**Issue by jessehu** [sigoden/dufs#324]

**Comment by sigoden** [comment]
If you don't want to edit the file, you may set read-only permissions.
Dufs allow you control the permissions of files or directorie with `--auth`.

**Comment by jessehu** [comment]
do you mean setting read-only permission for a single file via 'chmod 400'? Can it be done on dufs UI?

**Comment by sigoden** [comment]
Have you read the dufs readme? dufs can set permissions for paths. @jessehu 


![image](https://github.com/sigoden/dufs/assets/4012553/80368e4d-ea13-4811-9027-7f62c8122eb9)

**Comment by jessehu** [comment]
This solution requires restarting dufs for each file to be locked and unlocked. Not easy to use.

**Comment by sigoden** [comment]
dufs is a stateless file server and does not support file dynamic locking.

**Comment by jessehu** [comment]
I see. Just wondering if it can be archieved by:

1. is it possible to ignore deleting a file if its file mode is read only ?
2. or create/delete a `<filename>.lock` file when locking/unlocking this file on dufs UI?

**Comment by sigoden** [comment]
The only way I recommend to protect files is through auth. Other methods are error prone.

**Comment by jessehu** [comment]
Thanks. I tried the solution 1 and works well: 
![图片](https://github.com/sigoden/dufs/assets/10006/e5b61fa6-95e2-4b19-891c-662dbf7cc8ca)

**Comment by sigoden** [comment]
This solution is also error prone.

Here is the doc for `std::fs::Permissions`:
![image](https://github.com/sigoden/dufs/assets/4012553/b6f85f45-8ed4-41e3-ac77-24a246af7eea)

Permissions do not check ACLs. If the file permissions are "-rw-r--r-- ubuntu web..." and you run dufs in the "web" group, you may think dufs will think the file is readonly and return 403, but it's wrong, becase `permissions().readonly()` returns false and  dufs will return 500.

There is no easy way to determine whether a file is truely readonly for a process.

**Comment by jessehu** [comment]
Yup, it depends on which user used to run dufs, and this is controlled by the user who runs dufs.
